src/Kovan-DssSpell.t.sol

```diff
@@ -1835,86 +2122,275 @@ contract DssSpellTest is DSTest, DSMath {
+            ETH.approve(address(joinETHA), ETHIlkAmt);
+            joinETHA.join(address(this), ETHIlkAmt);
+            ETH.approve(address(joinETHB), ETHIlkAmt);
+            joinETHB.join(address(this), ETHIlkAmt);
+            ETH.approve(address(joinETHC), ETHIlkAmt);
+            joinETHC.join(address(this), ETHIlkAmt);
 
-        // Price is hacked, anyone can trip the breaker
-        clipMom.tripBreaker(address(clipYFIA));
+            WBTC.approve(address(joinWBTCA), WBTCIlkAmt / 10 ** (18 - 8));
+            joinWBTCA.join(address(this), WBTCIlkAmt / 10 ** (18 - 8));
 
-        assertEq(clipYFIA.stopped(), 2);
-    }
+            (,uint256 rate, uint256 spot,,) = vat.ilks("ETH-A");
+            vat.frob("ETH-A", address(this), address(this), address(this), int256(ETHIlkAmt), int256(mul(ETHIlkAmt, spot) / rate));
 
-    function testLerp() public {
-        LerpFabLike factory = LerpFabLike(addr.addr("LERP_FAB"));
+            (, rate, spot,,) = vat.ilks("ETH-B");
+            vat.frob("ETH-B", address(this), address(this), address(this), int256(ETHIlkAmt), int256(mul(ETHIlkAmt, spot) / rate));
 
-        assertEq(vow.hump(), 500 * RAD);
-        assertEq(factory.count(), 0);
+            (, rate, spot,,) = vat.ilks("ETH-C");
+            vat.frob("ETH-C", address(this), address(this), address(this), int256(ETHIlkAmt), int256(mul(ETHIlkAmt, spot) / rate));
 
-        vote(address(spell));
-        scheduleWaitAndCast(address(spell));
-        assertTrue(spell.done());
+            (, rate, spot,,) = vat.ilks("WBTC-A");
+            vat.frob("WBTC-A", address(this), address(this), address(this), int256(WBTCIlkAmt), int256(mul(WBTCIlkAmt, spot) / rate));
+        }
+
+        hevm.warp(block.timestamp + 1);
+        jug.drip("ETH-A");
+        jug.drip("ETH-B");
+        jug.drip("ETH-C");
+        jug.drip("WBTC-A");
+
+        uint256 auctionIdETHA = clipETHA.kicks() + 1;
+        uint256 auctionIdETHB = clipETHB.kicks() + 1;
+        uint256 auctionIdETHC = clipETHC.kicks() + 1;
+        uint256 auctionIdWBTCA = clipWBTCA.kicks() + 1;
+
+        dog.bark("ETH-A", address(this), address(this));
+        dog.bark("ETH-B", address(this), address(this));
+        dog.bark("ETH-C", address(this), address(this));
+        dog.bark("WBTC-A", address(this), address(this));
+
+        assertEq(clipETHA.kicks(), auctionIdETHA);
+        assertEq(clipETHB.kicks(), auctionIdETHB);
+        assertEq(clipETHC.kicks(), auctionIdETHC);
+        assertEq(clipWBTCA.kicks(), auctionIdWBTCA);
+
+        hevm.store(
+            address(end),
+            keccak256(abi.encode(address(this), uint256(0))),
+            bytes32(uint256(1))
+        );
+        assertEq(end.wards(address(this)), 1);
+
+        end.cage();
+        end.cage("ETH-A");
+        end.cage("ETH-B");
+        end.cage("ETH-C");
+        end.cage("WBTC-A");
+
+        (,,, address usr,,) = clipETHA.sales(auctionIdETHA);
+        assertTrue(usr != address(0));
```

Comment:
shouldn't this be changed to `usr == address(this)`? or what are we testing here?

Reply:
We are testing that the auction exists. I guess yours is valid as well.